### PR TITLE
[#84] Show message if there is empty data source

### PR DESCRIPTION
**Pull request by sivagollapalli**
#84

### CODE REVIEW

**Comment by ankane** [comment]
Hey @sivagollapalli, thanks for the PR 👍 

We'll need to make this an option so it's backward compatible. Also, this should happen after the data is remotely fetched and normalized.

**Comment by sivagollapalli** [comment]
@ankane Do you mean that we need to allow user able to set the message through some option something like `Chartkick.msg = "No data available"`. Am I right?

**Comment by ankane** [comment]
I'm thinking something like:

```js
new Chartkick.LineChart("chart-1", data, {emptyMessage: "No data available"})
```

Otherwise, it should do the previous behavior.

**Comment by sivagollapalli** [comment]
@ankane Added changes. Could you please check?

**Comment by ankane** [comment]
Hey @sivagollapalli, thanks for updating. However, it still doesn't happen after the data is normalized. You'll want to add it here: https://github.com/ankane/chartkick.js/blob/12be3a7546335b495b7173636a74d7df52fab4d3/chartkick.js#L1711

I've also added a few comments inline.

File: chartkick.js [hunk 1]
```diff
@@ -1666,7 +1666,24 @@
     return processSeries(chart, "bubble");
   }
 
+  function defaultEmptyMsg() {
```
Comment (ankane) [inline chartkick.js]: No need for a separate function.

File: chartkick.js [hunk 2]
```diff
@@ -1666,7 +1666,24 @@
...
+    return 'No data available';
+  }
+
   function createChart(chartType, chart, element, dataSource, opts, processData) {
+    if (Object.keys(dataSource).length == 0) {
```
Comment (ankane) [inline chartkick.js]: This needs to handle multiple series.

File: chartkick.js [hunk 3]
```diff
@@ -1666,7 +1666,24 @@
...
+      var emptyMsg;
+
+      if(opts['emptyMessage'] != undefined) {
+        emptyMsg = opts['emptyMessage']
+      } else {
+	emptyMsg = defaultEmptyMsg(); 
+      }
+
+      document.getElementById(element).innerHTML = '<center>' + emptyMsg + '</center>';
```
Comment (ankane) [inline chartkick.js]: There's a `setText` method to update text.

File: chartkick.js (hunk 1 above)
Comment (sivagollapalli) [inline chartkick.js]: Removed the function.

File: chartkick.js (hunk 3 above)
Comment (sivagollapalli) [inline chartkick.js]: Using `setText` function.

File: chartkick.js (hunk 2 above)
Comment (sivagollapalli) [inline chartkick.js]: Could you elaborate what is multiple series? Are you referring to these `[[]]`, `[{}, {}]` ?

**Comment by sivagollapalli** [comment]
@ankane Couldn't get the comment "However, it still doesn't happen after the data is normalized. You'll want to add it here: https://github.com/ankane/chartkick.js/blob/12be3a7546335b495b7173636a74d7df52fab4d3/chartkick.js#L1711". Could you explain?

**Comment by ldodds** [comment]
I could benefit from this PR being accepted, as I need the functionality. Happy to help move this forward.

@sivagollapalli my understanding is that @ankane is suggesting that you need to do the checks for an empty set of data after `processData` is called to check the normalised data?

@ankane is that the only fix required to get this merged?

**Comment by ankane** [comment]
@ldodds Another change that's needed is to do the current behavior if `emptyMessage` isn't specified (rather than a default message).

**Comment by ldodds** [comment]
@sivagollapalli just want to check to see if you're likely to move this forward or whether I should try and pick it up?

**Comment by ankane** [comment]
Cleaning up this PR due to inactivity, but happy to review another for this feature.

**Pull request closed without merging**